packages/vc-export/src/KiltRevocationStatusV1.ts

```diff
@@ -0,0 +1,100 @@
+ *
+ * @param credential The KiltCredentialV1 to which the status method is linked to.
+ * @param opts Additional parameters.
+ * @param opts.api An optional polkadot-js/api instance connected to the blockchain network on which the credential is anchored.
+ * If not given this function will try to retrieve a cached connection from the [[ConfigService]].
+ */
+export async function checkStatus(
+  credential: VerifiableCredential,
+  opts: { api?: ApiPromise } = {}
+): Promise<void> {
+  const { credentialStatus } = credential
+  if (credentialStatus?.type !== KILT_REVOCATION_STATUS_V1_TYPE)
+    throw new Error(
+      `credential must have a credentialStatus of type ${KILT_REVOCATION_STATUS_V1_TYPE}`
+    )
+  const { api = ConfigService.get('api') } = opts
+  const { assetNamespace, assetReference, assetInstance } =
+    assertMatchingConnection(api, credential)
+  if (assetNamespace !== 'kilt' || assetReference !== 'attestation') {
+    throw new Error(
+      `cannot handle revocation status checks for asset type ${assetNamespace}:${assetReference}`
+    )
+  }
+  if (!assetInstance) {
+    throw new Error(
+      'The CAIP-19 identifier must contain a token id decoding to the credential root hash'
+    )
+  }
+  const rootHash = base58Decode(assetInstance)
+  const encoded = await api.query.attestation.attestations(rootHash)
+  if (encoded.isNone)
+    throw new Error(
+      `Attestation data not found at latest block ${encoded.createdAtHash}`
+    )
+
+  const decoded = Attestation.fromChain(encoded, u8aToHex(rootHash))
+  const onChainCType = CType.hashToId(decoded.cTypeHash)
+  const delegationId = getDelegationNodeIdForCredential(credential)
+  if (
+    decoded.owner !== credential.issuer ||
+    onChainCType !== credential.credentialSchema.id ||
+    delegationId !== decoded.delegationId
+  ) {
+    throw new Error(
+      `Credential not matching on-chain data: issuer "${decoded.owner}", CType: "${onChainCType}"`
+    )
+  }
+  if (decoded.revoked !== false) {
+    throw new Error('Attestation revoked')
+  }
+}
+
+/**
+ * Creates a [[KiltRevocationStatusV1]] object from a credential hash and blochain identifier, which allow locating the credential's attestation record.
+ *
+ * @param chainIdOrGenesisHash The genesis hash (or CAIP-2 identifier) of the substrate chain on which the attestation record lives.
+ * @param rootHash The credential hash identifying the relevant attestation record on that chain.
+ * @returns A new [[KiltRevocationStatusV1]] object.
+ */
+export function fromGenesisAndRootHash(
+  chainIdOrGenesisHash: Caip2ChainId | U8aLike,
+  rootHash: U8aLike
+): KiltRevocationStatusV1 {
+  const chainId =
+    typeof chainIdOrGenesisHash === 'string' &&
+    chainIdOrGenesisHash.startsWith('polkadot')
+      ? chainIdOrGenesisHash
+      : Caip2.chainIdFromGenesis(u8aToU8a(chainIdOrGenesisHash))
+
+  return {
+    id: `${chainId}/kilt:attestation/${base58Encode(rootHash)}`,
```

Comment:
As mentioned in another comment, the namespace should be rather something generic, like `att` (since max 8 letters can be used), and the reference could be `KiltAttestationV1` or something like that. Great to have the identifier be the base58 encoding of the root hash.

Reply:
let's define a namespace on the specs repo. How you like using the pallet name as a namespace (truncated to 8 letters -> `attestat`) and the storage item (`attestations`) as the reference? That could be a general approach that works for practically anything on a substrate chain.

Reply:
I don't think that is the right direction. How would the pallet name be defined? By its crate name? But its runtime declaration? And then each chain would basically have its own namespace, while that should be left to references and not namespaces. I am in talks with Juan that manages the CAIP stuff, so let's hold on on this for a while longer. It might either make sense to have a generic namespace for each of ctype, attestation, delegation, and credential, or then a `kilt` namespace which would give us the freedom to define whatever within it.

Reply:
this discussion should move to https://github.com/KILTprotocol/spec-KiltCredentialV1/issues/7